mmv1/third_party/terraform/services/kms/resource_kms_crypto_key_test.go

```diff
@@ -172,6 +172,7 @@ func TestAccKmsCryptoKey_basic(t *testing.T) {
 					testAccCheckGoogleKmsCryptoKeyWasRemovedFromState("google_kms_crypto_key.crypto_key"),
 					testAccCheckGoogleKmsCryptoKeyVersionsDestroyed(t, projectId, location, keyRingName, cryptoKeyName),
 					testAccCheckGoogleKmsCryptoKeyRotationDisabled(t, projectId, location, keyRingName, cryptoKeyName),
+					resource.TestCheckResourceAttrSet("google_kms_crypto_key.crypto_key", "primary"),
```

Comment:
This is causing a failure because the key has been removed from this config - the check funcs run immediately after the config is applied.

```
 vcr_utils.go:152: Step 4/4 error: Check failed: Check 4/4 error: Not found: google_kms_crypto_key.crypto_key in [root]
```

Additionally, if it's possible to know ahead of time what the primary version's name is, it would be great to use [`TestCheckResourceAttr`](https://developer.hashicorp.com/terraform/plugin/sdkv2/testing/acceptance-tests/teststep#builtin-check-functions) to check the value instead of just whether or not there is one. (Sorry, I meant to recommend that func last time but said the wrong one.)

Reply:
This functionality is being added precisely because there's no way of knowing what the primary version's name is other than via this API output. 

Given that, what would you recommend here?

Reply:
`TestCheckResourceAttrSet` is correct in that case but it needs to be set on one of the earlier steps of the test.

Reply:
> This functionality is being added precisely because there's no way of knowing what the primary version's name is other than via this API output.

thanks 🤦  yep that makes sense

Reply:
I suppose for the purposes of the tests, you could generally assume it starts at 1 since the version isn't being updated and then construct the full name using the project/keyring name/key name (full name format is `projects/<project id>/locations/global/keyRings/keyring-example/cryptoKeys/crypto-key-example/cryptoKeyVersions/1)` -- I'm not seeing any examples of other checks like that though on a quick skim. Is that what you're suggesting?

Reply:
yes, I think that would be reasonable if it's possible to build that string easily! Otherwise just checking whether it's set should be fine. Just needs to happen on an earlier step either way.

Reply:
@melinath took a few tries but I think this should be good to go now